extensions/podman/packages/api/src/podman-extension-api.d.ts

```diff
@@ -0,0 +1,23 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { RunOptions, RunResult } from '@podman-desktop/api';
+
+export interface PodmanExtensionApi {
+  exec(args: string[], options?: RunOptions): Promise<RunResult>;
+}
```

Comment:
here I don't see how consumers will be able to use that API in case of multiple providers (applehv, libkrun)

For example I see an hardcoded 'podman' value in the implementation:

```ts
return execPodman(args, 'podman', options);
```

but if I look at execPodman method

```ts
export function execPodman(
  args: string[],
  containersProvider?: string,
  options?: extensionApi.RunOptions,
): Promise<extensionApi.RunResult> {
````

containersProvider is an option to specify like is it Apple Hypervisor, is it libkrun, etc.

Reply:
I think we should pass the object `ProviderContainerConnection` or `ContainerProviderConnection` instead of a string

If we pass `ProviderContainerConnection` podman extension could add a security checking that the providerId corresponds and we are not sending one registered by another extensions.

however;

`ContainerProviderConnection` would be easier as extension get this object pretty easily through, and is something used in the `BuildOptions`, `VolumeCreateOptions`, `pullImage`, `createNetwork` etc....

Reply:
I added additional optional parameter `connection: ProviderContainerConnection`, so clients can get instance using `provider.getContainerConnections` and use it to call exec.